Cinebase/Gateways/Container/LocalContainer.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  LocalContainer.swift
+//  Cinebase
+//
+//  Created by Gustavo Villar on 5/11/17.
+//  Copyright © 2017 Globant. All rights reserved.
+//
+
+import Foundation
+
+protocol LocalContainer: class {
+    
+    associatedtype Model
+    associatedtype ModelID
+    
+    func find(withID id: ModelID, completion: @escaping (Result<Model>) -> Void)
+    func add(_ model: Model, completion: @escaping (Result<Void>) -> Void)
+    func update(_ model: Model, completion: @escaping (Result<Void>) -> Void)
+    func delete(_ model: Model, completion: @escaping (Result<Void>) -> Void)
+    
+    func fetchAll(completion: @escaping (Result<[Model]>) -> Void)
```

Comment:
perhaps un fetch sortedByKey to have in the future some sort behavior

the find right now only by ID, what do you think to make it generic find(withValue value: , inKey key: String, completion...)

we considered 1 single cases, do you think that we will need group cases like 
find(withID, completion: Result<[Model]> for example filter by category or near us
fetch the same